### PR TITLE
Add retrospective ADR for usage of remote settings

### LINKED ISSUE #2309: Retrospective ADR for using Remote Settings to deploy
Add a retrospective ADR covering why Normandy deploys recipes with Remote Settings instead of using an independent transport.

### PR DESCRIPTION
**Pull request by mythmon**
This may help when reviewing recent problems with the signing infrastructure, and to better undrest
and Normandy's history.

Fixes #2309

### CODE REVIEW

**Comment by mythmon** [comment]
bors try

File: docs/adrs/adr-0002-remote-settings.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+* Good, because it reduces the amount of data transferred.
+* Good, because Normandy don't have to rely on a new outside services.
+* Bad, because it is a complicated system to implement on both server and client side.
+* Bad, because it re-invents a system already accessible in Firefox.
+
+3. Process recipes on the server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of download all recipes and filtering them locally, clients would
```
Comment (rehandalal) [inline docs/adrs/adr-0002-remote-settings.rst:105]: s/download/downloading

**Review by rehandalal — APPROVED** [review]
r+wc

**Comment by mythmon** [comment]
bors r= @rehandalal

**Pull request merged**